src/WinRT.Runtime/TypeNameSupport.cs

```diff
@@ -187,18 +187,27 @@ private static Type FindTypeByNameCore(string runtimeClassName, Type[] genericTy
             {
                 if (genericTypes != null)
                 {
+                    if (resolvedType == typeof(global::System.Nullable<>) && genericTypes[0].IsDelegate())
+                    {
+#pragma warning disable IL3050 // Generic type is reference type.

```

Comment:
Suppressing warnings like this will only affect the analyzer, but ILC will still produce warnings at compile time. That's also why the CI build failed (also, this way we could double check that those warnings do fail the build, nice 😄). We should probably move this code in the `if (genericTypes != null)` block into a small helper function, and add `[UnconditionalSuppressMessage]` over it for IL3050. That way ILC will also not warn for it.

Reply:
Since attribute suppressions have whole-method granularity, splitting out helper methods is a good practice. You can use local functions as well, if you don't want to add a method to the parent class.

Reply:
Yup, using local functions is the approach we've been using through CsWinRT already in similar situations (basically for all code paths only meant for non-AOT scenarios). It's quite convenient indeed and avoids cluttering types! 😄

Reply:
Addressed with local function.